

The interrupt routine will call both those functions even if the first doesn't
have any work to do.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/net/ixgb/ixgb_main.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletion(-)

diff -puN drivers/net/ixgb/ixgb_main.c~igxb-speedup drivers/net/ixgb/ixgb_main.c
--- 25/drivers/net/ixgb/ixgb_main.c~igxb-speedup	2004-08-30 00:50:40.179656168 -0700
+++ 25-akpm/drivers/net/ixgb/ixgb_main.c	2004-08-30 00:50:40.183655560 -0700
@@ -1614,8 +1614,12 @@ static irqreturn_t ixgb_intr(int irq, vo
 	}
 #else
 	for (i = 0; i < IXGB_MAX_INTR; i++)
-		if (!ixgb_clean_rx_irq(adapter) & !ixgb_clean_tx_irq(adapter))
+		if (ixgb_clean_rx_irq(adapter) == FALSE)
 			break;
+	for (i = 0; i < IXGB_MAX_INTR; i++)
+		if (ixgb_clean_tx_irq(adapter) == FALSE)
+			break;
+
 	/* if RAIDC:EN == 1 and ICR:RXDMT0 == 1, we need to
 	 * set IMS:RXDMT0 to 1 to restart the RBD timer (POLL)
 	 */
_
